plugins/inputs/socketstat/socketstat.go

```
@@ -0,0 +1,188 @@
+	// The first line is only headers
+	scanner.Scan()
+
+	flushData := false
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" {
+			continue
+		}
+		words := strings.Fields(line)
+
+		var err error
+		if !beginsWithBlank.MatchString(line) {
+			if flushData {
+				acc.AddFields(measurement, fields, tags)
+				flushData = false
+			}
+			// Delegate the real parsing to getTagsAndState, which manages various
+			// formats depending on the protocol
+			tags, fields = getTagsAndState(proto, words)
+			flushData = true
+		} else {
+			for _, word := range words {
+				if validValues.MatchString(word) {
+					kv := strings.Split(word, ":")
+					fields[kv[0]], err = strconv.ParseUint(kv[1], 10, 64)
+					if err != nil {
+						continue
+					}
+				}
+			}
+			acc.AddFields(measurement, fields, tags)
+			flushData = false
+		}
+	}
+	if flushData {
+		acc.AddFields(measurement, fields, tags)
+	}
+	return nil
+}
+
+func getTagsAndState(proto string, words []string) (map[string]string, map[string]interface{}) {
+	tags := map[string]string{}
+	fields := make(map[string]interface{})
+	tags["proto"] = proto
+	switch proto {
+	case "udp", "raw":
+		words = append([]string{"dummy"}, words...)
+	case "tcp", "dccp", "sctp":
+		fields["state"] = words[0]
+	}
+	switch proto {
+	case "tcp", "udp", "raw", "dccp", "sctp":
+		local := strings.Split(words[3], ":")
+		remote := strings.Split(words[4], ":")
+		tags["local_addr"] = strings.Join(local[:len(local)-1], ":")
+		tags["local_port"] = local[len(local)-1]
+		tags["remote_addr"] = strings.Join(remote[:len(remote)-1], ":")
+		tags["remote_port"] = remote[len(remote)-1]
+	case "unix", "packet":
+		fields["netid"] = words[0]
+		tags["local_addr"] = words[4]
+		tags["local_port"] = words[5]
+		tags["remote_addr"] = words[6]
+		tags["remote_port"] = words[7]
+	}
+	var err error
+	fields["recv_q"], err = strconv.ParseUint(words[1], 10, 64)
+	fields["send_q"], err = strconv.ParseUint(words[2], 10, 64)
+	if err != nil {
+	}
```

What is the idea behind this? If you want to ignore the error, why getting it in the first place? I think you really need to handle invalid values here.

---

It's now caught, and logged.